### PR TITLE
Update react testing library

### PR DESCRIPTION
**Pull request by snowystinger**
Closes 

Supports https://github.com/adobe/react-spectrum/pull/6437
Breaking changes outlined here https://github.com/testing-library/dom-testing-library/releases/tag/v10.0.0 where a new version of aria query was introduced which handles images without alt text better.

## ✅ Pull Request Checklist:

- [ ] Included link to corresponding [React Spectrum GitHub Issue](https://github.com/adobe/react-spectrum/issues).
- [ ] Added/updated unit tests and storybook for this change (for new code or code which already has tests).
- [ ] Filled out test instructions.
- [ ] Updated documentation (if it already exists for this component).
- [ ] Looked at the Accessibility Practices for this feature - [Aria Practices](https://www.w3.org/WAI/ARIA/apg/)

### CODE REVIEW

File: packages/@react-spectrum/test-utils/package.json [hunk 1]
```diff
@@ -28,7 +28,7 @@
     "@swc/helpers": "^0.5.0"
   },
   "peerDependencies": {
-    "@testing-library/react": "^13.0.0 || ^14.0.0",
+    "@testing-library/react": "^15.0.0",
```
Comment (snowystinger) [inline packages/@react-spectrum/test-utils/package.json:31]: given the difference in aria-query, i don't think we can support all these versions

File: package.json [hunk 2]
```diff
@@ -220,7 +220,8 @@
     "@parcel/rust": "2.12.1-dev.3224",
     "@parcel/types": "2.0.0-dev.1601",
     "@parcel/utils": "2.0.0-dev.1601",
-    "@parcel/workers": "2.0.0-dev.1601"
+    "@parcel/workers": "2.0.0-dev.1601",
+    "aria-query": "5.3.0"
```
Comment (snowystinger) [inline package.json:224]: have to resolve this so that we don't special case based on react version in the tests I had to fix

Comment (snowystinger) [reply package.json:224]: whelp, it didn't work
will work with @LFDanLu to see how we want to try to handle this

Comment (snowystinger) [reply package.json]: After chatting, we've decided to special case in the few tests which are affected

**Review by reidbarber — APPROVED** [review]

**Review by devongovett — APPROVED** [review]

**Pull request merged**